File: LR1/src/main/scala/comparator/IntegerComparator.scala
package comparator

import types.IntegerType

class IntegerComparator extends Comparator with Serializable {
  override def compare(o1: AnyRef, o2: AnyRef): Int = o1.asInstanceOf[IntegerType].getIntegerValue - o2.asInstanceOf[IntegerType].getIntegerValue
}

File: LR3/app/src/main/java/com/example/lr3/comparator/IntegerComparator.java
package com.example.lr3.comparator;

import com.example.lr3.types.IntegerType;

import java.io.Serializable;

public class IntegerComparator implements Comparator, Serializable {
    @Override
    public int compare(Object o1, Object o2) {
        return ((IntegerType) o1).getIntegerValue() - ((IntegerType) o2).getIntegerValue();
    }
}
